### ISSUE TITLE
Drop the `function a/1 does not match previous clause b/1` warning

### ISSUE THREAD

**Issue by yrashk** [elixir-lang/elixir#933]
I am starting another campaign to drop this warning:

``` elixir
iex(1)> defmodule A do
...(1)>   def a(1), do: 1
...(1)>   def b(1), do: 2
...(1)>   def a(2), do: 2
...(1)> end
iex:4: function a/1 does not match previous clause b/1
```

I think it is harmful in many ways. Not only we have situations when we do definitions with unquotes that often result in this warning (and require to split cycles into multiples of similar cycles to define those clauses together... which is really quite insane, not documented/explained anywhere), but it is also inherently harmful for writing gen_servers and alikes.

Instead of being able to write

``` elixir
def my_call, do: :gen_server.call(__MODULE__, :my_call)
def handle_call(:my_call, _, state), do: {:reply, :ok, state}

def my_call1, do: :gen_server.call(__MODULE__, :my_call1)
def handle_call(:my_call1, _, state), do: {:reply, :ok, state}
```

which is a way that would prevent a lot of mistakes from happening (forgetting to update the call or vice versa) and save time (not jumping up and down the source file), you would need to write really unproductive code like this:

``` elixir
def my_call, do: :gen_server.call(__MODULE__, :my_call)
def my_call1, do: :gen_server.call(__MODULE__, :my_call1)

def handle_call(:my_call, _, state), do: {:reply, :ok, state}
def handle_call(:my_call1, _, state), do: {:reply, :ok, state}
```

which is not really helpful at all. 

Bottom line: this warning is harmful. It provides little benefit and a lot of PITA.
